api/api.py

```diff
@@ -5,59 +5,66 @@
 from transport_co2 import estimate_co2, Mode
 
 
-def verify_coordinates_were_provided(
-    origin_lat, origin_lon, destination_lat, destination_lon
-):
+def verify_coordinates_were_provided(origin, destination):
     """Ensure all origin/destination coordinates were provided."""
-    return origin_lat and origin_lon and destination_lat and destination_lon
+    return (
+        origin
+        and destination
+        and origin["latitude"]
+        and origin["longitude"]
+        and destination["latitude"]
+        and destination["longitude"]
+    )
 
 
 def get_co2_estimate(
     transport_mode=None,
+    origin=None,
+    destination=None,
     distance_km=None,
     vehicle_occupancy=None,
-    origin_lat=None,
-    origin_lon=None,
-    destination_lat=None,
-    destination_lon=None,
+    *args,
+    **kwargs,
```

Comment:
Does this seem like we are getting wildcard list of parameters in the function, without any argument type control? Could we rework it so arguments to the functions will be more clear instead of `args`

Reply:
This is so the original request object will pass through unmodified.

Reply:
In this case, neither `args` nor `kwargs` are used in the function body, so we might want to change that. 😉 

Since the objective is to copy through any additional properties the leg might have, using `**kwargs` seems idiomatic and intuitive to me. On the other hand, `*args` doesn't do much here. Instead, I'd argue we should add `*` in front of the very first argument to force explicitly keyword-only arguments for all of them.